### PR TITLE
Polish for one-transaction view 

### PR DESCRIPTION
**Pull request by NikkiWines**
### Details

Follow up PR to https://github.com/Expensify/App/pull/36934 to handle some bugs and edge cases. 

### Fixed Issues
$ https://github.com/Expensify/App/issues/39512

### Tests

1. Log into an account that is a member of a workspace and make a single money request on the workspace chat
2. Navigate to the workspace chat and click on the expense report preview
3. Confirm that opens the report and that you see a combined expense view showing the details of both the expense report and transaction thread with an updated simplified icon (non-diagonal version) without a total.
4. Modify some elements of the expense like the category, description, etc. 
5. Confirm that the last text of the report in the LHN updates appropriately 

https://github.com/Expensify/App/assets/16747822/21903ef4-00bd-4cce-b0a0-8bf1758849d9

1. Create another money request on the workspace chat
2. Navigate back to the expense report and click into one of the transactions
3. Add a comment or two on that transaction
4. Go back to the report and delete the transaction that you had just commented on
5. Confirm you do _not_ see the combined one-transaction report view and instead see `[Deleted Request]` with some replies and the other transaction
6. Click into the replies and confirm they load correctly 

https://github.com/Expensify/App/assets/16747822/2ce0739e-0f78-446c-9a48-d9377bea1613

7. Delete the other transaction on the report and confirm you still see the standard report view with `[Deleted Request]` 

https://github.com/Expensify/App/assets/16747822/d2cdcc84-6f96-4ec4-beb6-382f63e735c3

- [x] Verify that no errors appear in the JS console

### Offline tests
N/A offline behavior not impacted 

### QA Steps

Same as test steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/16747822/b3c71c69-a45d-4a04-b5a2-50c4fb9fabd5

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/16747822/03fcac95-7966-4a89-b234-500acfd27c1f

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/16747822/cdbf11cb-56bd-4a75-8a7f-6b0f51d2fd8d

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/16747822/21903ef4-00bd-4cce-b0a0-8bf1758849d9

https://github.com/Expensify/App/assets/16747822/2ce0739e-0f78-446c-9a48-d9377bea1613

https://github.com/Expensify/App/assets/16747822/d2cdcc84-6f96-4ec4-beb6-382f63e735c3

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/16747822/2c7d20b2-e748-427f-a9d8-4d2b722e1292

https://github.com/Expensify/App/assets/16747822/8f3d1b9f-933e-45f6-aa73-5ea7b3c39299

</details>

### CODE REVIEW

**Comment by jayeshmangwani** [comment]
@allroundexperts, will you review this PR as you were the C+ for this [PR](https://github.com/Expensify/App/pull/36934) , Or should I review it?
cc: @NikkiWines

**Comment by allroundexperts** [comment]
I can review @jayeshmangwani

**Comment by NikkiWines** [comment]
Friendly bump on this @allroundexperts 🙇

**Comment by allroundexperts** [comment]
On it today!

**Comment by NikkiWines** [comment]
updated!

**Comment by allroundexperts** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by NikkiWines** [comment]
That's so weird, I'm entirely unable to reproduce that bug even though someone else has also reported it as an issue 🤔 I'll keep trying to reproduce locally so i can see where things are going awry

**Comment by NikkiWines** [comment]
I'm still unable to reproduce that currency / report total issue, but it's also been brought up before this PR [here](https://expensify.slack.com/archives/C049HHMV9SM/p1712165138133209?thread_ts=1712165004.169379&cid=C049HHMV9SM). So, @allroundexperts  let's move forward with this as it is and

**Comment by allroundexperts** [comment]
Hi @NikkiWines!

The LHN is not reflecting correctly if the report has been submitted or not in offline mode.

https://github.com/Expensify/App/assets/30054992/d03eba22-8073-48db-be16-f139be75fca9

**Comment by NikkiWines** [comment]
@allroundexperts I can reproduce that bug but it doesn't seem specific to the one-transaction report view, so let's not hold this PR on that change 🙇 

https://github.com/Expensify/App/assets/16747822/f829b1f4-22c8-42bb-b67d-90cdecddf113

**Comment by NikkiWines** [comment]
@allroundexperts friendly bump 🙇

**Review by allroundexperts — APPROVED** [review]
Looking good!

File: src/libs/ReportActionsUtils.ts
Comment (amyevans) [inline src/libs/ReportActionsUtils.ts:847]:
```suggestion
    // If there's only one IOU request action associated with the report but it's been deleted, then we don't consider this a oneTransaction report
```

**Review by amyevans — DISMISSED** [review]
LGTM! Just one NAB, feel free to merge

**Review by amyevans — APPROVED** [review]
Nice!

**Comment by kbecciv** [comment]
This PR is failing because of issue https://github.com/Expensify/App/issues/39450

The issue is reproducible in: All Platforms, didn't verify on mWeb yet.

https://github.com/Expensify/App/assets/93399543/49e9b1cc-3ee6-482e-92ba-2d01c4880fc5

https://github.com/Expensify/App/assets/93399543/a51884e7-612f-4d74-93e8-cce4380c304f

https://github.com/Expensify/App/assets/93399543/b3b0ba96-e76a-43d8-8b76-1119363c3fde

https://github.com/Expensify/App/assets/93399543/a9f6754d-5093-4ccb-b4c9-af6de10d1d78

**Comment by NikkiWines** [comment]
Weird, that's one of the things I tested it on right before merging. I'll look into fixing it but since it's the same behavior as before it's not a blocker 👍

File: src/libs/ReportActionsUtils.ts
Comment (brunovjk) [inline src/libs/ReportActionsUtils.ts:847]: When a user deleted a transaction that had replies, this condition caused the app to misidentify the single remaining transaction thread. As a result, the “Hold” and “Delete expense” options disappeared from the workspace chat menu. [Issue caused](https://github.com/Expensify/App/issues/64800).

**Pull request merged**
